### PR TITLE
[BEAM-2514] Improve error message on missing required value

### PR DESCRIPTION
**Pull request by manuzhang**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by jbonofre** [comment]
R: @jbonofre

**Comment by lukecwik** [comment]
This change assumes that all users are CLI users (compared to the existing code which assumed that all users were programmatic users). Should we word the message so its useful for both CLI and users who set the options programmatically?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsValidator.java [hunk 1]
```diff
@@ -46,6 +46,26 @@
...
+  /**
+   * Validates that the passed {@link PipelineOptions} from command line interface (CLI)
+   * conforms to all the validation criteria from the passed in interface.
+   *
+   * <p>Note that the interface requested must conform to the validation criteria specified on
+   * {@link PipelineOptions#as(Class)}.
+   *
+   * @param klass The interface to fetch validation criteria from.
+   * @param options The {@link PipelineOptions} to validate.
+   * @return The type
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsValidator.java]: Here and above.
nit: "The type" -> `Validated options.`

I'll fix this while merging.

**Comment by lukecwik** [comment]
R: @lukecwik

**Pull request closed without merging**